lib/dialects/postgres/connection-manager.js

```diff
@@ -153,26 +153,25 @@ class ConnectionManager extends AbstractConnectionManager {
         query += 'SELECT typname, oid, typarray FROM pg_type WHERE typtype = \'b\' AND typname IN (\'hstore\', \'geometry\', \'geography\')';
       }
 
-      return new Promise((resolve, reject) => {
-        connection.query(query)
-          .on('error', err => reject(err))
-          .on('row', row => {
-            let type;
-            if (row.typname === 'geometry') {
-              type = dataTypes.postgres.GEOMETRY;
-            } else if (row.typname === 'hstore') {
-              type = dataTypes.postgres.HSTORE;
-            } else if (row.typname === 'geography') {
-              type = dataTypes.postgres.GEOGRAPHY;
-            }
+      return new Promise((resolve, reject) => connection.query(query, (error, result) => error ? reject(error) : resolve([connection, result])));
```

Comment:
hmm, resolving with a tuple to maintain the connection is a bit weird. I didn't see that `connection` is still needed in scope, I would say in that case the nested promise is okay